rfcs/SchemaFragments.md

```diff
@@ -0,0 +1,79 @@
+# RFC: Schema Fragments
+
+## Problem
+
+When writing large schemas, there is no way to reuse similar fields between `ObjectTypeDefinition`(s) and `InputValueDefinition`(s). One can use [interfaces](https://spec.graphql.org/June2018/#sec-Interfaces) to enforce that particular fields are implemented, though this doesn't really help, as a schema creator I still have to repeat X fields on X amount of types.
+
+Below I have `Users`(s) and `Post`(s) and where both types have the following properties; `id`, `createdAt` and `updatedAt`:
+
+```graphql
+type User {
+  id: ID # Repeated
+  createdAt: DateTime # Repeated
+  updatedAt: DateTime # Repeated
+  name: String
+}
+
+type Post {
+  id: ID # Repeated
+  createdAt: DateTime # Repeated
+  updatedAt: DateTime # Repeated
+  content: String
+}
+```
+
+> Notice how the three properties are repeated.
+
+As mentioned, you can use interfaces here, so for example a `BaseInterface` that contains the properties, and then this is implemented on each type:
+
+```graphql
+interface BaseInterface {
+  id: ID # Repeated
+  createdAt: DateTime # Repeated
+  updatedAt: DateTime # Repeated
+}
+
+type User implements BaseInterface {
+  id: ID # Repeated
+  createdAt: DateTime # Repeated
+  updatedAt: DateTime # Repeated
+  name: String
+}
+
+type Post implements BaseInterface {
+  id: ID # Repeated
+  createdAt: DateTime # Repeated
+  updatedAt: DateTime # Repeated
+  content: String
+}
+```
+
+However, this isn't helpful at scale because your still repeating each field.
```

Comment:
Yes, this behaviour could be replicated if fields from interfaces are automatically added to Objects/Interfaces. About tooling, for sure it would help make it clearer and I'm keen to hear ideas.